packages/order-utils/src/asset_proxy_utils.ts

```diff
@@ -65,33 +66,46 @@ export const assetProxyUtils = {
                 }), but got ${assetProxyId}`,
             );
         }
-        const addressOffset = ERC20_PROXY_METADATA_BYTE_LENGTH - 1;
-        const encodedTokenAddress = encodedProxyMetadata.slice(0, addressOffset);
+        const addressOffset = ERC20_ASSET_DATA_BYTE_LENGTH - 1;
+        const encodedTokenAddress = encodedAssetData.slice(0, addressOffset);
         const tokenAddress = assetProxyUtils.decodeAddress(encodedTokenAddress);
-        const erc20ProxyData = {
+        const erc20AssetData = {
             assetProxyId,
             tokenAddress,
         };
-        return erc20ProxyData;
+        return erc20AssetData;
     },
-    encodeERC721ProxyData(tokenAddress: string, tokenId: BigNumber): string {
+    encodeERC721AssetData(tokenAddress: string, tokenId: BigNumber, receiverData?: string): string {
         const encodedAssetProxyId = assetProxyUtils.encodeAssetProxyId(AssetProxyId.ERC721);
         const encodedAddress = assetProxyUtils.encodeAddress(tokenAddress);
         const encodedTokenId = assetProxyUtils.encodeUint256(tokenId);
-        const encodedMetadata = Buffer.concat([encodedAddress, encodedTokenId, encodedAssetProxyId]);
-        const encodedMetadataHex = ethUtil.bufferToHex(encodedMetadata);
-        return encodedMetadataHex;
+        let encodedAssetData = Buffer.concat([encodedAddress, encodedTokenId]);
+        if (!_.isUndefined(receiverData)) {
+            const encodedReceiverData = ethUtil.toBuffer(receiverData);
+            const receiverDataLength = new BigNumber(encodedReceiverData.byteLength);
+            const encodedReceiverDataLength = assetProxyUtils.encodeUint256(receiverDataLength);
+            encodedAssetData = Buffer.concat([encodedAssetData, encodedReceiverDataLength, encodedReceiverData]);
+        }
+        encodedAssetData = Buffer.concat([encodedAssetData, encodedAssetProxyId]);
+        const encodedAssetDataHex = ethUtil.bufferToHex(encodedAssetData);
+        return encodedAssetDataHex;
     },
-    decodeERC721ProxyData(proxyData: string): ERC721ProxyData {
-        const encodedProxyMetadata = ethUtil.toBuffer(proxyData);
-        if (encodedProxyMetadata.byteLength !== ERC721_PROXY_METADATA_BYTE_LENGTH) {
+    decodeERC721AssetData(assetData: string): ERC721AssetData {
+        const encodedAssetData = ethUtil.toBuffer(assetData);
+        if (encodedAssetData.byteLength < ERC721_ASSET_DATA_BYTE_LENGTH) {
             throw new Error(
-                `Could not decode ERC20 Proxy Data. Expected length of encoded data to be 53. Got ${
-                    encodedProxyMetadata.byteLength
+                `Could not decode ERC20 Proxy Data. Expected length of encoded data to be at least 53. Got ${
+                    encodedAssetData.byteLength
                 }`,
             );
         }
-        const encodedAssetProxyId = encodedProxyMetadata.slice(-1);
+        const addressOffset = 0;
```

Comment:
I'd vote for turning these into constants at the top of the file like `ERC721_ASSET_DATA_BYTE_LENGTH`

Reply:
Agreed.